src/pocketmine/event/player/PlayerTransferEvent.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+declare(strict_types=1);
+
+namespace pocketmine\event\player;
+
+use pocketmine\event\Cancellable;
+use pocketmine\Player;
+
+class PlayerTransferEvent extends PlayerEvent implements Cancellable{
+	public static $handlerList = null;
+
+	protected $address;
+	protected $port = 19132;
+
```

Comment:
For example, another plugin would want to specifically cancel transfers from a particular plugin's particular action, but not others. This is an efficient way of interacting and sharing data with other plugins without direct dependency.

I have also considered a model like Android's Intent API, which allows actions to be taken even without knowing the specific existence of the other application (but for example, any applications that work with that particular type of data). This is however too big a change for the whole event system, so not gonna consider it in this pull request.